### ISSUE TITLE
okhttp-digest 1.14 error: java.lang.NullPointerException: Attempt to invoke interface method 'int java.util.Map.size()' on a null object reference

### ISSUE THREAD

**Issue by alexeyvasilyev** [rburgst/okhttp-digest#37]
Error in okhttp-digest 1.14. It works at the same time in okhttp-digest 1.13.

```
 W/System.err: java.lang.NullPointerException: Attempt to invoke interface method 'int java.util.Map.size()' on a null object reference
 W/System.err:     at java.util.HashMap.putMapEntries(HashMap.java:500)
 W/System.err:     at java.util.HashMap.<init>(HashMap.java:489)
 W/System.err:     at com.burgstaller.okhttp.digest.DigestAuthenticator.authenticateWithState(DigestAuthenticator.java:204)
 W/System.err:     at com.burgstaller.okhttp.DispatchingAuthenticator.authenticateWithState(DispatchingAuthenticator.java:55)
 W/System.err:     at com.burgstaller.okhttp.AuthenticationCacheInterceptor.intercept(AuthenticationCacheInterceptor.java:40)
 W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
 W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
 W/System.err:     at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185)
 W/System.err:     at okhttp3.RealCall.execute(RealCall.java:69)

```

**Comment by alexeyvasilyev** [comment]
Wireshark dumps
https://www.dropbox.com/s/8ulu6l76oj8jh4i/okhttp-digest-issues-37.zip?dl=0

**Comment by rburgst** [comment]
Thanks for your patch, this is then fixed in 1.15
